File: crudEmpleado-webcore/src/main/java/com/financieraOh/crudEmpleado_webcore/application/api/EmpleadoService.java
package com.financieraOh.crudEmpleado_webcore.application.api;

import com.financieraOh.crudEmpleado_webcore.application.dto.request.EmpleadoDTO;
import com.financieraOh.crudEmpleado_webcore.application.dto.response.ResponseDTO;
import com.financieraOh.crudEmpleado_webcore.domain.model.Empleado;

public interface EmpleadoService {

	public ResponseDTO<Empleado> listarEmpleados();

	public ResponseDTO<Empleado> obtenerEmpleado(String documentoId);

	public ResponseDTO<Empleado> registrarEmpleado(EmpleadoDTO empleado);

	public ResponseDTO<Empleado> modificarEmpleado(EmpleadoDTO empleado);

	public ResponseDTO<Empleado> eliminarEmpleado(String documentoId);
}

File: crudEmpleado-web/src/app/src/app/service/empleado.service.ts
import { HttpClient } from '@angular/common/http';
import { inject, Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { IEmpleadoTable } from '../models/empleadoTable';
import { IResponseData } from '../models/IResponseData';
import { IEmpleadoDTO } from '../models/empleadoDTO';

@Injectable({
  providedIn: 'root'
})
export class EmpleadoService {
  private apiUrl = 'http://127.0.0.1:8080/api/empleado';

  private http = inject(HttpClient);

  listarEmpleados(): Observable<IResponseData<IEmpleadoTable>> {
    return this.http.get<IResponseData<IEmpleadoTable>>(`${this.apiUrl}/getAll`);
  }

  obtenerEmpleado(documentoId: string): Observable<IResponseData<IEmpleadoDTO>> {
    return this.http.get<IResponseData<IEmpleadoDTO>>(`${this.apiUrl}/getDetalle/${documentoId}`);
  }

  registrarEmpleado(empleado: IEmpleadoDTO): Observable<any> {
    return this.http.post(`${this.apiUrl}/register`, empleado);
  }

  modificarEmpleado(empleado: IEmpleadoDTO): Observable<any> {
    return this.http.put(`${this.apiUrl}/update`, empleado);
  }

  eliminarEmpleado(documentoId: string): Observable<any> {
    return this.http.put(`${this.apiUrl}/delete/${documentoId}`, {});
  }
}